### DISCUSSION TITLE
Aircraft Not Trimming

### DISCUSSION THREAD

**Discussion by matheusdarocha** [JSBSim-Team/jsbsim#517] [Q&A]
Hello Fellas!

I have tried to trim this aircraft with Load Factor -1.
When I use Excel to find an alpha and elevator angle to trim, i find -6 deg and -3 deg respectivelly.
But when I run with JSBSim, the trim fails and I have no clue why.

Note that I already checked:
- Interpolation boundaries (all goes beyond -6 deg)
- Thrust Avaiable (there is enough thrust to maintain the flight at equilibrium)
- Speed and Altitude checked (also the Lift and Pitch considered at Excel are compatible with the JSBSim calculation)

The files can be downloaded at:
https://www.dropbox.com/s/0cab2k3hrknj38c/g6000%20%281%29.rar?dl=0

Thanks for now, waiting for some help!!!

**Comment by seanmcleod** [comment]
Maybe you're running into the range limits of AoA that the trim code tries, see - https://github.com/JSBSim-Team/jsbsim/discussions/428#discussioncomment-800949

**Comment by seanmcleod** [comment]
So the trimming code uses the following logic to determine what range of alpha to try, defaulting to `[-5, 20]`. So just short of the -6 degrees you calculated you need.

https://github.com/JSBSim-Team/jsbsim/blob/dd97e99032e6cce6ae3d0d91730f3c10f8ce94af/src/initialization/FGTrimAxis.cpp#L101-L107

The range can **optionally** be set using the `alphalimits` element in the aircraft config file.

https://github.com/JSBSim-Team/jsbsim/blob/dd97e99032e6cce6ae3d0d91730f3c10f8ce94af/src/models/FGAerodynamics.cpp#L339-L346

Or the limits can also be set by writing to the property tree, e.g.

```python
# Set alpha range for trim solutions
fdm['aero/alpha-max-rad'] = math.radians(12)
fdm['aero/alpha-min-rad'] = math.radians(-4.0)
```

**Comment by matheusdarocha** [comment]
Thanks a lot @seanmcleod it solved the problem!
